test/AspNet.Security.OAuth.Providers.Tests/VkId/VkIdTests.cs

```diff
@@ -19,7 +19,14 @@ public VkIdTests(ITestOutputHelper outputHelper)
 
     protected internal override void RegisterAuthentication(AuthenticationBuilder builder)
     {
-        builder.AddVkId(options => ConfigureDefaults(builder, options));
+        builder.AddVkId(
+            options =>
+            {
+                options.ClientId = "my-client-id";
+                options.ClientSecret = "my-client-secret";
+                options.Backchannel = CreateBackchannel(builder);
+                options.UseExchangeCodeAsyncStateValidation = false;
+            });
```

Comment:
Do something like this, then you only need to set one extra property:

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/34f683a19d855fdc133dbd176cc0ca9766f422b0/test/AspNet.Security.OAuth.Providers.Tests/Okta/OktaTests.cs#L15-L19

Why is it that you need to turn this off?

Reply:
I'm out of ideas how to mock IDataProtector to pass tests failing on this [line](https://github.com/HScokies/AspNet.Security.OAuth.Providers.Vkid/blob/366b9092112d7a08a4ca28f4cf39921973d99d63/src/AspNet.Security.OAuth.VkId/VkIdAuthenticationHandler.cs#L199) :(

Reply:
You'll need to configure the HTTP request interceptor to echo back the state in the response body in some way. For all the other providers this is handled [here](https://github.com/HScokies/AspNet.Security.OAuth.Providers.Vkid/blob/366b9092112d7a08a4ca28f4cf39921973d99d63/test/AspNet.Security.OAuth.Providers.Tests/Infrastructure/LoopbackRedirectHandler.cs#L83-L121) by adjusting the callback URL.

We shouldn't really be building in an option there's no real reason to ever turn off from a consumer's point of view _just_ to get the tests to pass.

Reply:
phew, looks like I've finally got it :D